transhumandesign/kag-base
# [fixed] issues related to not having a spectator team in a gamemode.

## Status

- **READY**
## Description

- This PR helps solve this issue. https://github.com/transhumandesign/kag-base/issues/430

- ~~Basically the same old behavior of !m when it comes to CTF and TTH servers.~~ (Same behavior for all gamemodes).

- When on any other gamemode, using !m switches you to the spectator team (even if it doesn't _officially exist_ for the gamemode itself).

- The visual moderation stuff works.

- Typing _!m_ again gets you back to your previous non-spectator team.

- To switch back from spectator to your old team, you should type !m, ~~since using the "change team" button doesn't work.~~ (now it does), or just use the normal "change team" button.

- ~~The KickAFK script no longer attempts to switch admins/mods to spectator (unless it's a CTF/TTH server), instead it shows in the chat a reminder that you can either use !m on servers without spectator teams to avoid being AFK, or simply leave and come back later.~~

- Now the kickAFK script does force-swap admins to spectator team.

## Steps to Test or Reproduce

- **AFK kick:**

1. Join a server, in-which you're an admin (or just /rcon /login <pass>; /rcon /reloadsecurity).

2. Remain AFK for enough time.

3. Get swapped to spectator with a message in chat stating that you can get back to your previous team using _!m_. 
[_The screenshot shows it working on a CTF server, but it would do the same with any other gamemode._]
![screen-19-03-13-15-18-09](https://user-images.githubusercontent.com/11915822/54286052-64860000-45a3-11e9-993c-bd809f8ceebd.png)

- **About the !m feature:**

1. Join a server having a gamemode that doesn't include a spectator team. (eg: sandbox)

2. Type !m in chat, you should be switched to spectators team.

3. Type !m when you're done, you should be back to your old team.
